torchao/dtypes/utils.py

```diff
@@ -28,38 +30,53 @@ def wrapper(*args, **kwargs):
         return func
     return decorator
 
+"""
+Base class for different LayoutType, should not be instantiated directly
+"""
+@dataclass(frozen=True)
+class LayoutType:
+    def pad_input(self, input: torch.Tensor) -> torch.Tensor:
```

Comment:
This is kind of misleading and should return a not implemented error in my opinion

The better way to do this is to add   `@abstractmethod` after importing `from abc import ABC, abstractmethod` so the error gets raised correctly

Reply:
not all layout needs padding though, why do we want to raise an error here?